api/src/index.js

```diff
@@ -11,21 +11,81 @@ app.use(bodyParser.json());
 
 const apiRouter = express.Router();
 
-// You can delete this route once you add your own routes
-apiRouter.get("/", async (req, res) => {
-  const SHOW_TABLES_QUERY =
-    process.env.DB_CLIENT === "pg"
-      ? "SELECT * FROM pg_catalog.pg_tables;"
-      : "SHOW TABLES;";
-  const tables = await knex.raw(SHOW_TABLES_QUERY);
-  res.json({ tables });
+// Respond with all meals in the future (relative to the when datetime)
+app.get("/future-meals", async (req, res, next) => {
+  try {
+    const now = new Date().toISOString();
+    const futureMeals = await knex("meal").where("when", ">", now);
+    res.json(futureMeals);
+  } catch (error) {
+    next(error);
+  }
+});
+
+// Respond with all meals in the past (relative to the when datetime)
+app.get("/past-meals", async (req, res, next) => {
+  try {
+    const now = new Date().toISOString();
+    const pastMeals = await knex("meal").where("when", "<", now);
+    res.json(pastMeals);
+  } catch (error) {
+    next(error);
+  }
+});
+
+// Respond with all meals sorted by ID
+app.get("/all-meals", async (req, res, next) => {
+  try {
+    const allMeals = await knex("meal").orderBy("id");
+    res.json(allMeals);
+  } catch (error) {
+    next(error);
+  }
+});
+
+// Respond with the first meal (meaning with the minimum id)
+app.get("/meals/first-meal", async (req, res, next) => {
+  try {
+    const firstMeal = await knex("meal").orderBy("id").first();
+    if (!firstMeal) {
+      const error = new Error("No meals found");
+      error.status = 404;
+      return next(error);
+    }
+    res.json(meal);
+  } catch (error) {
+    next(error);
+  }
+});
+
+// Respond with the last meal (meaning with the maximum id)
+app.get("/meals/last-meal", async (req, res, next) => {
```

Comment:
The prefix `/meals` is not part of the requirements

```suggestion
app.get("/last-meal", async (req, res, next) => {
```